File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] arr) {
        int maxp=0;
        int minp=Integer.MAX_VALUE;
        for(int i=0;i<arr.length;i++){
            minp=Math.min(minp,arr[i]);
            maxp=Math.max(arr[i]-minp,maxp);
        }
        return maxp;
        
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit=0;
        int loss=INT_MAX;
        for(int i=0;i<prices.size();i++){
            loss=min(loss,prices[i]);
            profit=max(prices[i]-loss,profit);
        }
        return profit;
        
    }
};